### DISCUSSION TITLE
Why change the Text property of TextInputEventArgs from set to Init?

### DISCUSSION THREAD

**Discussion by ZSYMAX** [AvaloniaUI/Avalonia#11888] [Q&A]
I hope to modify the value of Text when TextEntering. Is there any other way to do this?

**Comment by timunie** [comment]
Please explain your usecase a bit more. What exactly do you want to achieve?

**Reply by soerendd** [reply]
I have the same issue. I have a behavior which connects to TextInputEvent and wants to validate input against some rules and if they do not match it should change the text

**Reply by Gillibald** [reply]
You can always use TextBox.TextChangingEvent to achieve what you want
